lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -55,7 +55,7 @@ def initialize(cmd, debug)
         # wait for the pipe server to signal ready, and fail if no response in 10 seconds
 
         # wait up to 30 seconds in 0.2 second intervals to be able to open the pipe
-        150.times do
+        (pipe_timeout * 5).times do
```

Comment:
Not sure if this makes sense, but maybe...

```ruby 
sleep_interval = 0.2
(pipe_timeout / sleep_interval).times do

# ...

sleep sleep_interval
```

Reply:
Does `times` work with a float? I got errors when I tried but didn't investigate too closely in an earlier local iteration of this.

Reply:
Float doesn't have the .times method. Only int. You could do `(pipe_timeout / sleep_interval).to_i.times` and that will work with some imprecision to avoid errors.